lib/net/ip.go

```diff
@@ -42,6 +42,12 @@ func (i *IP) UnmarshalJSON(b []byte) error {
 	return i.UnmarshalText([]byte(s))
 }
 
+// ParseIP returns an IP from a string
```

Comment:
It's nice to keep the behavior consistent with the `net` package.  We should just be making sure to do the nil check when we call this function.